src/nacl/hash.py

```diff
@@ -17,10 +17,80 @@
 import nacl.bindings
 import nacl.encoding
 
+BYTES = nacl.bindings.crypto_generichash_BYTES
+BYTES_MIN = nacl.bindings.crypto_generichash_BYTES_MIN
+BYTES_MAX = nacl.bindings.crypto_generichash_BYTES_MAX
+KEYBYTES = nacl.bindings.crypto_generichash_KEYBYTES
+KEYBYTES_MIN = nacl.bindings.crypto_generichash_KEYBYTES_MIN
+KEYBYTES_MAX = nacl.bindings.crypto_generichash_KEYBYTES_MAX
+SALTBYTES = nacl.bindings.crypto_generichash_SALTBYTES
+PERSONALBYTES = nacl.bindings.crypto_generichash_PERSONALBYTES
+
+_b2b_hash = nacl.bindings.crypto_generichash_blake2b_salt_personal
+
 
 def sha256(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha256(message))
 
 
 def sha512(message, encoder=nacl.encoding.HexEncoder):
     return encoder.encode(nacl.bindings.crypto_hash_sha512(message))
+
+
+def generichash(data, digest_size=BYTES, key=b'',
```

Comment:
While for my own expected use cases I only need the salted and personalized version, I think feature-parity of the bindings with the C library can aid those coming to the python bindings from libsodium's documentation, which refers to the blake2 name (and API) only in a page about key derivation.